src/mq2.c

```diff
@@ -0,0 +1,120 @@
+/***** Includes *****/
+#include "adc.h"
+#include "mq2.h"
+#include "timers.h"
+#include "math.h"
+#include <stdio.h>
+
+#ifdef STM32F1
+#include "stm32f1xx_hal.h"
+#endif
+#ifdef STM32F4
+#include "stm32f4xx_hal.h"
+#include "stm32f429xx.h"
+#endif
+
+
+/***** Defines *****/
+#define RL      5000        /* Total load resistance in kohm */
+#define ROUT    3000        /* Voltage divider output resistance */
+
+/***** Local consts *****/
+const double LPG_GAS[2] = {-0.47, 1.291};
+const double CO_GAS[2] = {-0.34, 1.502};
+const double SMOKE_GAS[2] = {-0.44, 1.542};
+const double* MQ2_GAS[3] = {LPG_GAS, CO_GAS, SMOKE_GAS};
```

Comment:
W sumie spoko
Alternatywnie można zrobić:
#define LPG_GAS {-0.47, 1.291}
...
const double MQ2_GAS[3][2] = {LPG_GAS}

Zgaduję, że inspirowane API. Różnica jest taka, że tam są struktury z różnymi dziwnymi rzeczami, a tu jest zwykła tabela stałych. Ale nie ma to chyba większego znaczenia na koniec dnia. Może jakbyś optymalizował pamięć, to oszczędzisz kilkadziesiąt bajtów robiąc "po mojemu"